File: test/bot_sources/haskell/basic_bot.hs
module Main where
import Control.Monad
import System.IO
import System.Random

randomRPS = do
  randInt <- randomRIO (0, 3 :: Int)
  if randInt == 0 then do putStrLn "rock";
  else do
    if randInt == 1 then do putStrLn "paper";
    else do putStrLn "scissors"

process (x:xs) = do
  case x of
    "action" -> do randomRPS
    _        -> return ()

loop = do
    line <- getLine
    process (words line)
    eof <- isEOF
    unless eof loop

main = loop

File: test/bot_sources/pascal/basic_bot.pas
//__main__

program RPS;

{$ifdef fpc}{$mode delphi}{$H+}{$endif}
{$ifndef fpc}{$apptype console}{$endif}

uses
  Classes, SysUtils;

procedure Run;
var
    s              : string;
    done           : Boolean;
    st             : TStringList;
    act            : string;
    MyRandomNumber : integer;
begin
    done := false;
    st:=TStringList.Create;
    try
        st.Delimiter:=#32; // space
        while not eof(input) do begin
            readln(s);

            if s = '' then Continue;
            st.DelimitedText:=s;
            if st.Count=0 then Continue;

            act := st[0];

            if act = 'action' then begin

                MyRandomNumber := random(3);
                if MyRandomNumber = 0 then writeln('rock')
                else if MyRandomNumber = 1 then writeln('paper')
                else if MyRandomNumber = 2 then writeln('scissors')

            end;
        if done then break;
        end;
    finally
        st.Free;
    end;
end;

begin
  run();
end.
